src/topology/subset_properties.lean

```diff
@@ -437,31 +437,47 @@ is_clopen_inter hs (is_clopen_compl ht)
 
 end clopen
 
-section irreducible
+section preirreducible
 
-/-- An irreducible set is one where there is no non-trivial pair of disjoint opens. -/
-def is_irreducible (s : set α) : Prop :=
+/-- An preirreducible set is one where there is no non-trivial pair of disjoint opens. -/
+def is_preirreducible (s : set α) : Prop :=
 ∀ (u v : set α), is_open u → is_open v →
-  (∃ x, x ∈ s ∩ u) → (∃ x, x ∈ s ∩ v) → ∃ x, x ∈ s ∩ (u ∩ v)
+  (s ∩ u).nonempty → (s ∩ v).nonempty → (s ∩ (u ∩ v)).nonempty
+
+/-- An irreducible set is one that is nonempty and
```

Comment:
I think you should explain this definition by giving a precise reference to the literature where the importance of nonemptiness is stressed. And also refer to this in the definition of `is_preirreducible`.

Reply:
Same thing below for connected, also.

Reply:
Done.